port/unix/omrsysinfo.c

```diff
@@ -7557,3 +7557,86 @@ omrsysinfo_get_number_context_switches(struct OMRPortLibrary *portLibrary, uint6
 	return OMRPORT_ERROR_SYSINFO_NOT_SUPPORTED;
 #endif /* defined(LINUX) */
 }
+
+/*
+ * Get the process ID and commandline for each process.
+ * @param[in] portLibrary The port library.
+ * @param[in] callback The function to be invoked for each process with the process ID and command info.
+ * @param[in] userData Data passed to the callback.
+ * @return 0 on success, or the first non-zero value returned by the callback.
+ */
+uintptr_t
+omrsysinfo_get_processes(struct OMRPortLibrary *portLibrary, OMRProcessInfoCallback callback, void *userData)
+{
+#if defined(LINUX)
+	struct dirent *entry = NULL;
```

Comment:
Locals should be declared in the smallest block possible; `entry` should be declared where it is initialized (on line 7588).
Only these need to be declared here:
```c
	uintptr_t callback_result = 0;
	DIR *dir = opendir("/proc");
```